src/api/VoteMonitor.Api.Form/Handlers/OptionQueriesHandler.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using AutoMapper;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using VoteMonitor.Api.Form.Models;
+using VoteMonitor.Api.Form.Queries;
+using VoteMonitor.Entities;
+
+namespace VoteMonitor.Api.Form.Handlers
+{
+	public class OptionQueriesHandler : IRequestHandler<FetchAllOptionsCommand, List<OptionDto>>,
+										IRequestHandler<GetOptionByIdCommand, OptionDto>,
+										IRequestHandler<AddOptionCommand, OptionDto>,
+										IRequestHandler<UpdateOptionCommand, int>
+
+	{
+		private readonly VoteMonitorContext _context;
+		private readonly IMapper _mapper;
+		private readonly ILogger _logger;
+
+		public OptionQueriesHandler(VoteMonitorContext context, IMapper mapper, ILogger logger)
+		{
+			_context = context;
+			_mapper = mapper;
+			_logger = logger;
+		}
+
+		public async Task<List<OptionDto>> Handle(FetchAllOptionsCommand request, CancellationToken cancellationToken)
+		{
+			return await _context.Options.Select(x => _mapper.Map<OptionDto>(x)).ToListAsync(cancellationToken);
+		}
+
+		public Task<OptionDto> Handle(GetOptionByIdCommand request, CancellationToken cancellationToken)
+		{
+			var option = _context.Options.FirstOrDefault(c => c.Id == request.OptionId);
+
+			var optionDto = _mapper.Map<OptionDto>(option);
+
+			return Task.FromResult(optionDto);
+		}
+
+		public async Task<OptionDto> Handle(AddOptionCommand request, CancellationToken cancellationToken)
+		{
+			var optionEntity = _mapper.Map<Option>(request.Option);
+
```

Comment:
The id is not auto-generated.
Currently adding a new Option throws the following error:
`Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while updating the entries. See the inner exception for details. ---> System.Data.SqlClient.SqlException: Cannot insert the value NULL into column 'Id', table 'mv-local-dev.dbo.Options'; column does not allow nulls. INSERT fails.`

Reply:
My bad, ran on a database that was not updated with auto generated ids.